### PR TITLE
[feat-19] Prefect로 작업 관리 및 모델 업데이트

### PR DESCRIPTION
**Pull request by lazychoi**
tasks.py
- db에서 데이터 불러오기
- 전처리: 불필요한 컬럼 삭제, 결측치 처리, 인코딩, 정규화
- 모델 학습
- mlflow로parameters, metrics, model 저장
- metric(auc)이 기존 production된 모델보다 높으면 교체

train_flow.py :
- 위 tasks를 실행

issue : #19

### CODE REVIEW

**Review by pastnfast — APPROVED** [review]
LGTM

File: train_prefect/tasks.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        -1,
+    )
+
+    return x_data, y_data, ct
+
+
+@task(log_stdout=True, nout=2)
+def train_model(model, x, y):
+
+    x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2, stratify=y)
```
Comment (AstroNaut000) [inline train_prefect/tasks.py:115]: x_train,**x_test**,y_train,**y_test**에서 test 부분을 val로 변경

File: train_prefect/train_flow.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    "objective": "binary:logistic",
+    "max_depth": 6,
+    "learning_rate": 0.1,
+    "n_estimators": 100,
+    "n_jobs": -1,
+}
+model = XGBClassifier(use_label_encoder=False, **params)
+model_name = model.__class__.__name__
+
+with Flow("Model Training Flow") as flow:
```
Comment (Ilevk) [inline train_prefect/train_flow.py]: Flow 이름은 식별이 가능하도록 변경

File: train_prefect/train_flow.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    eval_metric = Parameter("Evaluation Metric", "auc")
+    df = get_data()
+    x, y, preprocessed_model = preprocessing(df)
```
Comment (Ilevk) [inline train_prefect/train_flow.py]: preprocessed_model도 동일하게 mlflow에 log

File: train_prefect/tasks.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+@task(log_stdout=True)
+def change_production_model(model_name, current_version, eval_metric):
+
+    production_model = None
+
+    client = MlflowClient()
+    current_model = client.get_model_version(model_name, current_version)
+
+    name_filter = f"name='{model_name}'"
+    client = MlflowClient()
```
Comment (Ilevk) [inline train_prefect/tasks.py]: 두번 정의할 필요가 없음

File: train_prefect/tasks.py [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+    precision_score,
+    recall_score,
+    f1_score,
+    roc_auc_score,
+)
+
+
+@task(log_stdout=True)
+def get_data():
+    url = "postgresql://postgres:postgres@localhost:5432/hoteldb"
```
Comment (Ilevk) [inline train_prefect/tasks.py]: 접속 정보가 모두 드러난다 어떻게 숨길 수 있을까?

File: train_prefect/train_flow.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    model, metrics = train_model(model, x, y)
+    current_version = log_model(model, model_name, params, metrics, eval_metric)
+    production_version = change_production_model(
+        model_name, current_version, eval_metric
+    )
+flow.run()
```
Comment (Ilevk) [inline train_prefect/train_flow.py]: flow.register 해보기

**Review by AstroNaut000 — APPROVED** [review]
LGTM
